packages/frontend/src/hooks/echarts/useEcharts.ts

```diff
@@ -753,6 +769,82 @@ const getEchartAxis = ({
     };
 };
 
+const getValidStack = (series: EChartSeries | undefined) => {
+    return series && (series.type === 'bar' || !!series.areaStyle)
+        ? series.stack
+        : undefined;
+};
+
+const calculateStackTotal = (
+    row: ResultRow,
+    series: EChartSeries[],
+    flipAxis: boolean | undefined,
+) => {
+    return series.reduce<number>((acc, s) => {
+        const hash = flipAxis ? s.encode?.x : s.encode?.y;
+        if (hash && row[hash]?.value.raw) {
+            acc += Number(row[hash]?.value.raw);
+        }
+        return acc;
+    }, 0);
+};
+
+const getStackTotalRows = (
+    rows: ResultRow[],
+    series: EChartSeries[],
+    flipAxis: boolean | undefined,
+): [unknown, unknown, number][] => {
+    return rows.map((row) => {
+        const total = calculateStackTotal(row, series, flipAxis);
+        const hash = flipAxis ? series[0].encode?.x : series[0].encode?.y;
+        if (!hash) {
+            return [null, null, 0];
+        }
+        return flipAxis
+            ? [0, row[hash]?.value.raw, total]
+            : [row[hash]?.value.raw, 0, total];
+    });
+};
+
+// To hack the stack totals in echarts we need to create a fake series with the value 0 and display the total in the label
```

Comment:
ahhh, that is clever. I think this is what's causing this behaviour https://github.com/lightdash/lightdash/pull/3744#issuecomment-1309883389